fenicsprecice/fenicsprecice.py

```diff
@@ -273,9 +288,60 @@ def initialize(self, coupling_subdomain, mesh, function_space, write_function=No
+            write_function_space = write_object.function_space()
+            write_function = write_object
+        elif type(write_object) is FunctionSpace:  # preCICE will use default zero values for initialization.
+            write_function_space = write_object
+            write_function = None
+        elif write_object is None:
+            pass
+        else:
+            raise Exception("Given write object is neither of type dolfin.functions.function.Function or "
+                            "dolfin.functions.functionspace.FunctionSpace")
+
+        if read_function_space is None and write_function_space:
+            self._coupling_type = CouplingMode.UNIDIR_WRITE
+            assert (self._config.get_write_data_name())
+            print("Participant {} is write-only participant".format(self._config.get_participant_name()))
+            mesh = write_function_space.mesh()
+        elif read_function_space and write_function_space is None:
+            self._coupling_type = CouplingMode.UNIDIR_READ
+            assert (self._config.get_read_data_name())
+            print("Participant {} is read-only participant".format(self._config.get_participant_name()))
+            mesh = read_function_space.mesh()
+        elif read_function_space and write_function_space:
+            self._coupling_type = CouplingMode.BIDIR
+            assert (self._config.get_read_data_name() and self._config.get_write_data_name())
+            mesh = read_function_space.mesh()
+        elif read_function_space is None and write_function_space is None:
+            raise Exception("Neither read_function_space nor write_function_space is provided. Please provide a write "
+                            "function_space if one-way coupling with this participant only writing data is intended. "
+                            "Please provide a read_function_space if one-way coupling with this participant only "
+                            "reading data is intended. If two-way coupling is implemented then both read_function_space"
+                            " and write_object need to be provided.")
+        else:
+            raise Exception("Incorrect read and write function space combination provided. Please check input "
+                            "parameters in initialization")
+
+        if self._coupling_type is CouplingMode.UNIDIR_READ or self._coupling_type is CouplingMode.BIDIR:
+            self._read_function_type = determine_function_type(read_function_space)
+            self._read_function_space = read_function_space
+
+        if self._coupling_type is CouplingMode.UNIDIR_WRITE or self._coupling_type is CouplingMode.BIDIR:
+            # Ensure that function spaces of read and write functions are defined using the same mesh
+            self._write_function_type = determine_function_type(write_function_space)
+
+        coords = self._read_function_space.tabulate_dof_coordinates()
```

Comment:
I just ran into the following error:

```
Participant HeatNeumann is write-only participant
Traceback (most recent call last):
  File "heat.py", line 130, in <module>
    precice_dt = precice.initialize(coupling_boundary, write_object=u_D_function)
  File "/home/benjamin/.local/lib/python3.8/site-packages/fenicsprecice/fenicsprecice.py", line 336, in initialize
    coords = self._read_function_space.tabulate_dof_coordinates()
AttributeError: 'NoneType' object has no attribute 'tabulate_dof_coordinates'
---[precice]  Implicitly finalizing in destructor
```

Used the following modified version of the HT tutorial: 
[l_336_error.txt](https://github.com/precice/fenics-adapter/files/5607978/l_336_error.txt)

Note: you can use `git apply l_336_error.txt` to apply the provided changes.